roles/sap_swpm/tasks/swpm/prepare_software.yml

```diff
@@ -81,6 +137,15 @@
   register: sap_swpm_sapcar_file_name_stat
   failed_when: not sap_swpm_sapcar_file_name_stat.stat.exists
 
+- name: SAP SWPM Pre Install - Ensure correct permissions and ownership of the SAPCAR*EXE file
+  ansible.builtin.file:
+    path: "{{ sap_swpm_sapcar_path }}/{{ sap_swpm_sapcar_file_name }}"
+    recurse: no
+    mode: "{{ sap_swpm_files_sapcar_mode }}"
+    owner: "{{ sap_swpm_files_sapcar_owner }}"
+    group: "{{ sap_swpm_files_sapcar_group }}"
+  when: sap_swpm_set_file_permissions
```

Comment:
I miss the sap_swpm_sapcar_path definition, I guess is somewhere else, but is it not necessary to check that it is defined? or do we take it for granted? (same on task 85)

Reply:
The parameter `sap_swpm_sapcar_path` is contained in defaults/main.yml and needs to be defined in the playbook or inventory.
It's also checked in the task named `SAP SWPM Pre Install - Check availability of SAPCAR path` (line 83). In the roles `sap_hana_install` and `sap_install_media_detect`, we search for the SAPCAR file but a similar mechanism has not yet been implemented for `sap_swpm`. So we better check if the file and directory is available and modify the permissions of both.
If the `sap_swpm_sapcar_path` (the directory where sapcar is located is not available, the task `Check availability of SAPCAR path` (line 83) will fail.
And if the SAPCAR file is not there, the previous task, `Check availability of SAPCAR` (line 138), will fail.

Reply:
Ok, makes sense!